lib/tx_build/signer_key.ex

```diff
@@ -51,16 +49,20 @@ defmodule Stellar.TxBuild.SignerKey do
     signer_type = SignerKeyType.new(:SIGNER_KEY_TYPE_PRE_AUTH_TX)
 
     key
-    |> (&:crypto.hash(:sha256, &1)).()
+    |> KeyPair.raw_pre_auth_tx()
     |> UInt256.new()
     |> SignerKey.new(signer_type)
   end
 
-  @spec validate_signer_type(type :: type()) :: validation()
-  defp validate_signer_type(type) when type in ~w(ed25519 sha256_hash pre_auth_tx)a,
-    do: {:ok, type}
-
-  defp validate_signer_type(_type), do: {:error, :invalid_signer_type}
+  @spec get_signer_type(key :: String.t()) :: validation()
```

Comment:
@CristhianRodriguezMolina @FelipeGuzmanSierra 

1. Could you please define a new @type for "keys".
2. What if I set a key like `GTX`. is it a valid key? Should we validate the string length as well?

Reply:
1. Ok, we are gonna add it.
2. It is not necessary to validate the length, because the function `validate_signer_key/1` validates every signature decoding it with the `StrKey` `decode/2` function through the `KeyPair` module. So with the `KeyPair.validate_public_key("GTX")` we will get a `{:error, :invalid_ed25519_public_key}` response.